src/groundStation/groundStation.py

```diff
@@ -151,22 +151,48 @@ def transaction(self, server, port, buf):
             return {}
         libcsp.send(conn, buf)
         libcsp.buffer_free(buf)
+        rxDataList = []
         packet = libcsp.read(conn, 10000)
         if packet is None:
             print('packet is None; no more packets')
             return
 
         data = bytearray(libcsp.packet_get_data(packet))
         length = libcsp.packet_get_length(packet)
-        rxData = self.parser.parseReturnValue(
+        rxDataList.append(self.parser.parseReturnValue(
             libcsp.conn_dst(conn),
             libcsp.conn_src(conn),
             libcsp.conn_sport(conn),
             data,
-            length)
-        if rxData is None:
+            length))
+        
+        if rxDataList is None:
             print('ERROR: bad response data')
-        return rxData
+            return
+
+        #code following is specific to housekeeping multi-packet transmission
+        if libcsp.conn_src(conn) != 1 or libcsp.conn_sport(conn) != 9 or data[0] != 0 or len(data) < 3 or data[2] != 1:
```

Comment:
Let's use the port definitions from `system.py`

Reply:
all done